src/Core/Infrastructure/SemanticVersionHelper.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+/**
+ * Copyright © 2016-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+declare(strict_types=1);
+
+namespace Core\Infrastructure;
+
+class SemanticVersionHelper
+{
+    /**
+     * @param string $semanticVersion
+     *
+     * @return int|null
+     */
+    public static function getMajorVersion(string $semanticVersion): ?int
+    {
+        if (!static::isSemanticVersion($semanticVersion)) {
+            return null;
+        }
+
+        $versionParts = explode('.', $semanticVersion);
+        if (!$versionParts) {
+            return null;
+        }
+
+        return (int)array_shift($versionParts);
+    }
+
+    /**
+     * @param string $version
+     *
+     * @return bool
+     */
+    protected static function isSemanticVersion(string $version): bool
+    {
+        return preg_match('/^\d+\.\d+\.\d+$/', $version) === 1;
```

Comment:
Semantic version can contain also the stability suffix - stable|RC|beta|alpha|dev . And it is used in Spryker, eg.g. propel package.

In this case regex won't work. I think it's better to use `composer/semver` if possible.

Reply:
thanks replaced with `return Comparator::greaterThanOrEqualTo($version, '0.0.1') === true;`